# Calculating GDP per Capita for US counties in 2019
Ravi Shankar
March 1, 2022


# Importing County data

The excel file was downloaded [here](), and then every column but the GDP in 2019 was dropped, along with the county/state names


In [46]:
import pandas as pd

#This imports the Real Gross Domestic Product*, by U.S. County, 2017–2020 in
#Thousands of chained (2012) dollars
# *Real means inflation adjusted

#GDP Data Frame
GDPDF = pd.read_csv(r'CountyGDP.csv')
GDPDF

Unnamed: 0            2019
0                                         United States  19,032,672,000
1                                                   NaN             NaN
2                                               Alabama     203,383,898
3                                               Autauga       1,540,762
4                                               Baldwin       7,134,734
...                                                 ...             ...
3217                                           Washakie         349,686
3218                                             Weston         322,576
3219  1. On January 2, 2019, two new county equivale...             NaN
3220  2. Virginia combination areas consist of one o...             NaN
3221           Source. U.S. Bureau of Economic Analysis             NaN

[3222 rows x 2 columns]

Now we do some data cleaning, removing NaNs. This drops 56 rows, making the file from 3222 rows to 3166 rows

In [47]:

#Drop NaN rows (spaces and footnotes)
GDPDF = GDPDF.dropna()

#Add Column Names
GDPDF.columns = ["CTYNAME", "GDP2019" ]

GDPDF = GDPDF.reset_index()


GDPDF

index        CTYNAME         GDP2019
0         0  United States  19,032,672,000
1         2        Alabama     203,383,898
2         3        Autauga       1,540,762
3         4        Baldwin       7,134,734
4         5        Barbour         729,105
...     ...            ...             ...
3161   3214     Sweetwater       3,677,972
3162   3215          Teton       2,268,742
3163   3216          Uinta         881,052
3164   3217       Washakie         349,686
3165   3218         Weston         322,576

[3166 rows x 3 columns]

# Importing County Population

The excel file was downloaded [here](https://www.census.gov/programs-surveys/popest/technical-documentation/research/evaluation-estimates/2020-evaluation-estimates/2010s-counties-total.html), and then every column but the Population estimate in 2019 was dropped, along with the county/state names.


In [48]:
import pandas as pd



#Population Data Frame
popDF = pd.read_csv(r'CountyPopulation.csv')

popDF

STATE  COUNTY            CTYNAME  POPESTIMATE2019
0         1       0            Alabama          4907965
1         1       1     Autauga County            55769
2         1       3     Baldwin County           223565
3         1       5     Barbour County            24657
4         1       7        Bibb County            22313
...     ...     ...                ...              ...
3189     56      37  Sweetwater County            42917
3190     56      39       Teton County            23385
3191     56      41       Uinta County            20196
3192     56      43    Washakie County             7824
3193     56      45      Weston County             6880

[3194 rows x 4 columns]

Now we do some data cleaning. There are 3194 rows in the population data and only 3166 in the 

In [49]:

#Remove state and county FIPS code for now
popDF = popDF[["CTYNAME", "POPESTIMATE2019"]]

popDF

CTYNAME  POPESTIMATE2019
0               Alabama          4907965
1        Autauga County            55769
2        Baldwin County           223565
3        Barbour County            24657
4           Bibb County            22313
...                 ...              ...
3189  Sweetwater County            42917
3190       Teton County            23385
3191       Uinta County            20196
3192    Washakie County             7824
3193      Weston County             6880

[3194 rows x 2 columns]

# Remove State Names



In [50]:
#import numpy as np
from numpy import loadtxt

#read text file into NumPy array
#data = loadtxt('my_data.txt')

# Join County Population and GDP Data


In [56]:
#GDP per Capita Data Frame = GDP_per_cap_DF
GDP_per_cap_DF = pd.concat([GDPDF,popDF], axis=1)



# Default value of display.max_rows is 10 i.e. at max 10 rows will be printed.
# Set it None to display all rows in the dataframe
pd.set_option('display.max_rows',None)

GDP_per_cap_DF



index                                       CTYNAME         GDP2019  \
0        0.0                                 United States  19,032,672,000   
1        2.0                                       Alabama     203,383,898   
2        3.0                                       Autauga       1,540,762   
3        4.0                                       Baldwin       7,134,734   
4        5.0                                       Barbour         729,105   
5        6.0                                          Bibb         380,453   
6        7.0                                        Blount         932,215   
7        8.0                                       Bullock         247,229   
8        9.0                                        Butler         615,641   
9       10.0                                       Calhoun       4,114,829   
10      11.0                                      Chambers         817,533   
11      12.0                                      Cherokee         527,358   
12      13.0                                       Chilton         980,365   
13      14.0                                       Choctaw         492,926   
14      15.0                                        Clarke         736,710   
15      16.0                                          Clay         419,318   
16      17.0                                      Cleburne         274,512   
17      18.0                                        Coffee       1,315,749   
18      19.0                                       Colbert       2,341,000   
19      20.0                                       Conecuh         466,540   
20      21.0                                         Coosa         176,482   
21      22.0                                     Covington       1,100,109   
22      23.0                                      Crenshaw         428,320   
23      24.0                                       Cullman       2,649,646   
24      25.0                                          Dale       2,522,516   
25      26.0                                        Dallas       1,102,126   
26      27.0                                        DeKalb       1,857,673   
27      28.0                                        Elmore       1,809,270   
28      29.0                                      Escambia       1,206,764   
29      30.0                                        Etowah       2,766,192   
30      31.0                                       Fayette         321,519   
31      32.0                                      Franklin         926,671   
32      33.0                                        Geneva         526,336   
33      34.0                                        Greene         218,149   
34      35.0                                          Hale         241,685   
35      36.0                                         Henry         407,934   
36      37.0                                       Houston       4,957,322   
37      38.0                                       Jackson       1,381,755   
38      39.0                                     Jefferson      41,244,762   
39      40.0                                         Lamar         305,015   
40      41.0                                    Lauderdale       2,367,072   
41      42.0                                      Lawrence         501,518   
42      43.0                                           Lee       5,603,187   
43      44.0                                     Limestone       3,572,973   
44      45.0                                       Lowndes         377,807   
45      46.0                                         Macon         478,726   
46      47.0                                       Madison      23,670,284   
47      48.0                                       Marengo         681,595   
48      49.0                                        Marion         792,988   
49      50.0                                      Marshall       3,276,199   
50      51.0                 

In [ ]:
They don't match up perfectly so we have to edit them down now:

In [53]:
# EDIT manually  the data to get them to match
GDPDF = GDPDF.drop("United States")

GDPDF

KeyError: "['United States'] not found in axis"